conans/client/build/mesonx.py

```diff
@@ -0,0 +1,375 @@
+import os
+import platform
+from configparser import ConfigParser
+from pathlib import Path
+
+from conans.client import tools
+from conans.client.build import defs_to_string, join_arguments
+from conans.client.build.cppstd_flags import cppstd_from_settings
+from conans.client.toolchain.meson import MesonMachineFile, MesonToolchain, MesonDefaultToolchain
+from conans.client.tools.env import environment_append, _environment_add, no_op
+from conans.client.tools.oss import args_to_string
+from conans.errors import ConanException
+from conans.model.build_info import DEFAULT_BIN, DEFAULT_INCLUDE, DEFAULT_LIB
+from conans.model.version import Version
+from conans.util.files import decode_text, get_abs_path, mkdir
+from conans.util.runners import version_runner
+
+# Differences from stock Meson integration:
+# - does not override 'default_library' if no 'shared' options was set
+# - saves `build_subdir` in __init__
+# - cleaned up interface: removed old arguments, removed build_dir and etc
+# - added machine file support
+# - added machine file generation for `toolchain`
+# - backend-agnostic meson methods are used by default with ninja as a fallback for older meson versions.
+# - env variables with dependency search paths are not appended when executing meson commands
+# - requires pkg-config generator
+# - install_folder is always added to pkg-config search paths now (since pc generator is required now)
+
+# TODO:
+# - add pkg_config_path to configure properly (cross scenario and native) or move it to machine file generation
+# - fix env variables reading for native\cross once it's possible (i.e when <https://github.com/conan-io/conan/issues/7091> is fixed)
+# - cleanup comments
+# - add tests
+# - Can we use Py3.x? meson requires Python 3.5+
+# - deal with typings (remove? add everywhere?)
+
+# Questions:
+# - Should we keep `cross-file`/`native-file` kwarg or should they be moved to `options` instead?
+# - Same q as ^ for pkg-config-paths
+# - How should the user extend `_get_cpu_family_and_endianness_from_arch` in case he has a custom unknown arch?
+# - Should we generate machine files in dump() or in __init__()?
+# - Move `MesonDefaultToolchainGenerator` to `MesonDefaultToolchain`?
+
+class MesonX():
+    def __init__(self, conanfile, build_subdir: str = None, backend: str = None, append_vcvars: bool = False) -> None:
```

Comment:
Please remove all type annotations. We are not using them yet, and they are completely breaking Python2.7 clients. It is fine if the inner functionality only works with Python 3, but the introduction of this cannot break legacy Python 2.7, even if they are not using Meson: https://conan-ci.jfrog.info/blue/organizations/jenkins/ConanTestSuite/detail/PR-7155/2/pipeline

CI should be green, the tests using this functionality could be using ``@unittest.skipIf(six.PY2,...``

Reply:
I see, will be fixed. I'm a Python novice so this is all new to me =)

Reply:
Fixed.